draft-beck-tls-trust-anchor-ids.md

```diff
@@ -387,15 +387,15 @@ In most X.509 deployments, a compromise of _any_ root CA's private key compromis
 
 A multi-certificate deployment model avoids these transition problems. Key rotation may proceed as follows:
 
```

Comment:
Ah, good catch. Some overall comments:

For brevity, I think we can take it as read that if you generate a new root CA, you generate a new ID if you're participating in this mechanism.

We've actually never defined the client to having a preference between trust anchors. The client *can*, but it's a little goofy because only the server knows the sizes of the proofs and the performance of the corresponding keys, which is why we have a server preference list. Also, at the end of the day, the client doesn't care as long as it got a proof from _something_ it trusts.

The other complexity is that this draft covers both `certificate_authorities`-style selection and the client-triggered one. But I think we can say what we're trying to say more succinctly and more generally.

How about this?

1. The CA operator generates a new root CA with a separate key, but continues operating the old root CA.
2. Root programs being trusting the new root CA alongside the old one, as a transition state. Both roots are configured to participate in this mechanism.
3. When subscribers request certificates, the CA issues certificates from both roots and provisions the subscriber with both certificates.
4. Trust anchor negotiation selects the old root for relying parties that only trust the old root, and the new root for relying parties that only trust the new root. In transition, relying parties that trust both may be served either, but prioritizing the new root helps monitor the transition.
5. Once subscribers have been provisioned with new certificates, root programs can safely distrust the old root in new relying parties. The CA operator may continue to operate the old root CA for as long as it wishes to serve subscribers that, in turn, wish to serve older relying parties.

Reply:
Yeah, the generic language works just fine here.  Took it.